### PR TITLE
fix: return to initial state when reseting hash in Accordion & Tabs #11100

### LINKED ISSUE #11100: Tabs: Browser back doesn't take us back to default tab
#### How to reproduce this bug:

1. Setup some tabs
2. Set these options: `data-deep-link="true"` and  `data-update-history="true"`. The 2nd option pushes state onto the browser history. 
3. Click a few random tabs
4. Click the browser back button and watch the URL bar
5. Repeat step 4 multiple times

The hash changes as you click the tabs. When clicking back the state reverses and the previous tab loads accordingly. However, when getting back to the original URL, the tab state doesn't update. 

* Landing page -> example.com/  -> active Tab 1
* Click Tab 2 -> example.com/#tab-2 -> active Tab 2
* Click browser back -> example.com/ -> active Tab 2 (BUG)

#### Browser(s) and Device(s) tested on:

Desktop Chrome

#### Foundation Version(s) you are using:

6.4.3

#### Test case link:

https://codepen.io/anon/pen/LdeRmW

### PR DESCRIPTION
**Pull request by ncoden**
## Description

When deep-linking is enabled and the user goes back in the history up to when no hash is set, the active panel or tab does not change and is still the one that was selected the first.

This pull request make Accordion and Tabs displaying the panel or tab that was active at the component initialization.

### Changes:
- Save up the active panel/tab at Accordion/Tabs initialization (`_initialAnchor`)
- Display the initial panel/tab when the hash change to `""`.

- Closes https://github.com/zurb/foundation-sites/issues/11100
- Replaces [#11101 - feat: add class for active tab when no hash is set in Tabs ](https://github.com/zurb/foundation-sites/issues/11101) (@jamiechong)

## Types of changes

- [ ] Documentation
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (anything that would change an existing functionality)
- [ ] Maintenance (refactor, code cleaning, development tools...)

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] The pull request title and template are correctly filled.
- [x] The pull request targets the right branch (`develop` or `develop-v...`).
- [x] My commits are correctly titled and contain all relevant information.
- ~~I have updated the documentation accordingly to my changes (if relevant).~~
- ~~I have added tests to cover my changes (if relevant).~~

### CODE REVIEW

**Review by SassNinja — CHANGES_REQUESTED** [review]
@ncoden I've tested the accordion and it works – but only if an item is initially active!

If no item has initially the class `is-active` is fails.
I think we should cover that case, too.

File: js/foundation.tabs.js [hunk 1]
```diff
@@ -98,8 +104,14 @@ class Tabs extends Plugin {
...
+      if (!anchor.length && this._initialAnchor) {
+        anchor = this._initialAnchor;
+      }
+
       //need a hash and a relevant anchor in this tabset
-      if(anchor.length) {
+      if (anchor.length) {
         var anchorNoHash = (anchor.indexOf('#') >= 0 ? anchor.slice(1) : anchor);
         var $link = this.$element.find(`[href$="${anchor}"],[data-tabs-target="${anchorNoHash}"]`).first();
         if ($link.length) {
```
Comment (SassNinja) [inline js/foundation.tabs.js]: In the course of testing the tabs I noticed the focus style remains when using history back.

tabs_deep_linking_480px_low

What do you think about auto focusing the link to fix this?

```javascript
if ($link.length) {
  $link.focus();
  this.selectTab($(anchor), true);
```

**Review by SassNinja — CHANGES_REQUESTED** [review]

File: js/foundation.tabs.js (hunk 1 above)
Comment (ncoden) [reply js/foundation.tabs.js]: This is an other issue, not related to this pull request. Why not, but I wonder what's the impact on accessibility to move the focus like this.

**Comment by ncoden** [comment]
@SassNinja I refactored the component to clean up a bit the logic and split the options processing/component logic from simple DOM manipulations. It should resolve the first bug you pointed out.

**Comment by SassNinja** [comment]
> Could you review this again please ? :)

It's working better now: history back adjust the adressbar as expected.

But the active state is not working as expected:
when I start without an initially opened tab, click on a tab and go back in history, I don't get the original active state (none tab opened). Instead the previously clicked tab is active (although the page url has changed).

Shall I provide an example repo or is it clear what I mean?

**Comment by ncoden** [comment]
> Shall I provide an example repo or is it clear what I mean?

I understood the issue, and through I resolved this issue. If you look at the code, you can see that `_closeAll()` is called if we go back to a state where no link is active.

**Comment by ncoden** [comment]
Actually, I fixed this issue for the Accordion only. :p

**Comment by SassNinja** [comment]
Just to be clear: the accordion component works as expected.
The issues described above only refers to the tabs components.

Here's a quick gist with the content of my `\test\visual\tabs\deep-linking.html`
https://gist.github.com/SassNinja/c194da0830de81a510061c99ffcb8938

**Comment by ncoden** [comment]
> Just to be clear: the accordion component works as expected.
> The issues described above only refers to the tabs components.

Yeah I got it. I need to do the same refactor in Tabs as I did in Accordion to resolve it. I'll take care of this sunday.

**Comment by ncoden** [comment]
@SassNinja Done.

**Comment by ncoden** [comment]
@SassNinja 👋 Could you test it again?

**Comment by SassNinja** [comment]
@ncoden lgtm 👍 

Just two things don't look nice
- focus state doesn't get reseted on history back (as mentioned earlier)
- the hash tag in the browser url doesn't get removed when I toggle a tab, so that no panel is opened (when using `data-active-collapse="true"`)

But nothing related to this PR – so good to merge imo.

**Review by SassNinja — APPROVED** [review]

**Pull request merged**